### PR TITLE
Crypto.updates

### PR DESCRIPTION
**Pull request by hbs**

### CODE REVIEW

File: warp10/src/main/java/io/warp10/script/WarpScriptLib.java [hunk 1]
```diff
@@ -615,9 +625,19 @@
   public static final String HASH = "HASH";
   public static final String TOSSSS = "->SSSS";
   public static final String SSSSTO = "SSSS->";
+  public static final String MD2 = "MD2";
+  public static final String MD4 = "MD4";
   public static final String MD5 = "MD5";
   public static final String SHA1 = "SHA1";
   public static final String SHA256 = "SHA256";
+  public static final String SHA224 = "SHA224";
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/WarpScriptLib.java:633]: That would be perfect with `"SHA224"` preceding `"SHA256"`.

File: warp10/src/main/java/io/warp10/script/WarpScriptLib.java [hunk 2]
```diff
@@ -1461,9 +1481,20 @@
...
     addNamedWarpScriptFunction(new DIGEST(SHA1, SHA1Digest.class));
     addNamedWarpScriptFunction(new DIGEST(SHA256, SHA256Digest.class));
+    addNamedWarpScriptFunction(new DIGEST(SHA224, SHA224Digest.class));
+    addNamedWarpScriptFunction(new DIGEST(SHA384, SHA384Digest.class));
+    addNamedWarpScriptFunction(new DIGEST(SHA512, SHA512Digest.class));
+    addNamedWarpScriptFunction(new DIGEST(RIPEMD128, RIPEMD128Digest.class));
+    addNamedWarpScriptFunction(new DIGEST(RIPEMD160, RIPEMD160Digest.class));
+    addNamedWarpScriptFunction(new DIGEST(RIPEMD256, RIPEMD256Digest.class));
+    addNamedWarpScriptFunction(new DIGEST(RIPEMD320, RIPEMD320Digest.class));
+    addNamedWarpScriptFunction(new DIGEST(GOST, GOST3411Digest.class));
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/WarpScriptLib.java:1496]: Seems there are also Tiger and Whirlpool digests available. I don't know it there are worth to to be added, though.

File: warp10/src/main/java/io/warp10/script/WarpScriptLib.java [hunk 3]
```diff
@@ -1461,9 +1481,20 @@
 
     // Crypto functions
     addNamedWarpScriptFunction(new HASH(HASH));
+    addNamedWarpScriptFunction(new DIGEST(MD2, MD2Digest.class));
+    addNamedWarpScriptFunction(new DIGEST(MD4, MD4Digest.class));
     addNamedWarpScriptFunction(new DIGEST(MD5, MD5Digest.class));
     addNamedWarpScriptFunction(new DIGEST(SHA1, SHA1Digest.class));
     addNamedWarpScriptFunction(new DIGEST(SHA256, SHA256Digest.class));
+    addNamedWarpScriptFunction(new DIGEST(SHA224, SHA224Digest.class));
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/script/WarpScriptLib.java:1489]: Same here, `SHA224` before `SHA256` would make the code much more appealing to the eye ;).

**Review by ftence — CHANGES_REQUESTED** [review]

**Comment by hbs** [comment]
PTAL

**Review by ftence — APPROVED** [review]

**Pull request merged**